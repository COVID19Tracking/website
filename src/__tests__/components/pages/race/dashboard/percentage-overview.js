import React from 'react'
import renderer from 'react-test-renderer'
import PercentageOverview from '~components/pages/race/dashboard/percentage-overview'

describe('Components : Race : Dashboard : Percentage Overview', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <PercentageOverview
          stateName="New Mexico"
          dataType="race"
          testPercent={0.27}
          hospitalizationPercent={0.993}
          casePercent={0.22}
          deathPercent={0.043}
          className="race-class"
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()

    const nullTree = renderer
      .create(
        <PercentageOverview
          stateName="New Mexico"
          dataType="race"
          className="race-class"
        />,
      )
      .toJSON()
    expect(nullTree).toMatchSnapshot()

    const nullTestHospDeathTree = renderer
      // Null values for death, test, and hospitalization values.
      .create(
        <PercentageOverview
          stateName="New Mexico"
          dataType="race"
          casePercent={0.22}
          className="race-class"
        />,
      )
      .toJSON()
    expect(nullTestHospDeathTree).toMatchSnapshot()
  })
})
