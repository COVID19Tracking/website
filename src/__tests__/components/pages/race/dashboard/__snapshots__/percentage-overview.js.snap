// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Race : Dashboard : Percentage Overview renders correctly 1`] = `
<div
  className="race-class"
>
  <h3
    className="title"
  >
    New Mexico
     has reported 
    race
     data for:
  </h3>
  <div
    className="data"
  >
    <div
      className="dataItem"
    >
      <p>
        <span
          className="percent"
        >
          22
          %
        </span>
         
        <span
          className="a11y-only"
        >
          of
        </span>
         
        <span
          className="percentCaption"
        >
          Cases
        </span>
      </p>
    </div>
    <div
      className="dataItem"
    >
      <p>
        <span
          className="percent"
        >
          4
          %
        </span>
         
        <span
          className="a11y-only"
        >
          of
        </span>
         
        <span
          className="percentCaption"
        >
          Deaths
        </span>
      </p>
    </div>
    <div
      className="dataItem"
    >
      <p>
        <span
          className="percent"
        >
          27
          %
        </span>
         
        <span
          className="a11y-only"
        >
          of
        </span>
         
        <span
          className="percentCaption"
        >
          Tests
        </span>
      </p>
    </div>
    <div
      className="dataItem"
    >
      <p>
        <span
          className="percent"
        >
          99
          %
        </span>
         
        <span
          className="a11y-only"
        >
          of
        </span>
         
        <span
          className="percentCaption"
        >
          Hospitalizations
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`Components : Race : Dashboard : Percentage Overview renders correctly 2`] = `null`;

exports[`Components : Race : Dashboard : Percentage Overview renders correctly 3`] = `
<div
  className="race-class"
>
  <h3
    className="title"
  >
    New Mexico
     has reported 
    race
     data for:
  </h3>
  <div
    className="data"
  >
    <div
      className="dataItem"
    >
      <p>
        <span
          className="percent"
        >
          22
          %
        </span>
         
        <span
          className="a11y-only"
        >
          of
        </span>
         
        <span
          className="percentCaption"
        >
          Cases
        </span>
      </p>
    </div>
    <div
      className="dataItem"
    >
      <p>
        <span
          className="percent"
        >
          <img
            alt="Alert icon"
            src="test-file-stub"
          />
        </span>
         
        <span
          className="a11y-only"
        >
          of
        </span>
         
        <span
          className="percentCaption"
        >
          Deaths
        </span>
      </p>
    </div>
    <div
      className="dataItem"
    >
      <p>
        <span
          className="percent"
        >
          <img
            alt="Alert icon"
            src="test-file-stub"
          />
        </span>
         
        <span
          className="a11y-only"
        >
          of
        </span>
         
        <span
          className="percentCaption"
        >
          Tests
        </span>
      </p>
    </div>
    <div
      className="dataItem"
    >
      <p>
        <span
          className="percent"
        >
          <img
            alt="Alert icon"
            src="test-file-stub"
          />
        </span>
         
        <span
          className="a11y-only"
        >
          of
        </span>
         
        <span
          className="percentCaption"
        >
          Hospitalizations
        </span>
      </p>
    </div>
  </div>
</div>
`;
